From adb88b875a0486a9edacd6ede9527b158163d22a Mon Sep 17 00:00:00 2001
From: Govindraj Raja <Govindraj.Raja@imgtec.com>
Date: Wed, 4 Nov 2015 14:16:37 +0000
Subject: io: adc: cc10001: Add dev pm ops

Adc can be gated during system wide suspend,
so add dev ops for adc to gate and ungate adc clocks
during suspend and resume.

Change-Id: I7739814e2667d2ee98f393918c77d09c39441965
Signed-off-by: Govindraj Raja <Govindraj.Raja@imgtec.com>
---
 drivers/iio/adc/cc10001_adc.c | 28 ++++++++++++++++++++++++++++
 1 file changed, 28 insertions(+)

diff --git a/drivers/iio/adc/cc10001_adc.c b/drivers/iio/adc/cc10001_adc.c
index 8254f52..a3b9fec 100644
--- a/drivers/iio/adc/cc10001_adc.c
+++ b/drivers/iio/adc/cc10001_adc.c
@@ -426,6 +426,33 @@ static int cc10001_adc_remove(struct platform_device *pdev)
 	return 0;
 }
 
+
+#ifdef CONFIG_PM_SLEEP
+static int cc10001_adc_suspend(struct device *dev)
+{
+	struct iio_dev *indio_dev = dev_get_drvdata(dev);
+	struct cc10001_adc_device *adc_dev = iio_priv(indio_dev);
+
+	clk_disable(adc_dev->adc_clk);
+
+	return 0;
+}
+
+static int cc10001_adc_resume(struct device *dev)
+{
+	struct iio_dev *indio_dev = dev_get_drvdata(dev);
+	struct cc10001_adc_device *adc_dev = iio_priv(indio_dev);
+
+	clk_enable(adc_dev->adc_clk);
+
+	return 0;
+}
+#endif /* CONFIG_PM_SLEEP */
+
+static const struct dev_pm_ops cc10001_adc_pm_ops = {
+	SET_SYSTEM_SLEEP_PM_OPS(cc10001_adc_suspend, cc10001_adc_resume)
+};
+
 static const struct of_device_id cc10001_adc_dt_ids[] = {
 	{ .compatible = "cosmic,10001-adc", },
 	{ }
@@ -435,6 +462,7 @@ MODULE_DEVICE_TABLE(of, cc10001_adc_dt_ids);
 static struct platform_driver cc10001_adc_driver = {
 	.driver = {
 		.name   = "cc10001-adc",
+		.pm	= &cc10001_adc_pm_ops,
 		.of_match_table = cc10001_adc_dt_ids,
 	},
 	.probe	= cc10001_adc_probe,
-- 
1.9.1

